content/en/apps/guides/data/analytics/building-dbt-models.md

```diff
@@ -0,0 +1,233 @@
+One additional model will be needed for each form, and for any aggreagations, dashboards, or reuseable views which use those form responses as input.
+If using the [configurable contact hierarchy]({{< ref "apps/reference/app-settings/hierarchy#app_settingsjson-contact_types" >}}), it may also be useful to add models for other contact types.
+
+## Setup
+
+CHT Sync uses a [DBT package](https://docs.getdbt.com/docs/build/packages) to run models, which is configured by providing a URL to a github repository.
+If it is necessary for application specific models to be private, the [cht-pipeline repository](https://github.com/medic/cht-pipeline/) can be forked into a private repository.
+If they can be public, a branch of the public repository can be created.
+
+The url is set in the `CHT_PIPELINE_BRANCH_URL` environment variable, either in .env if using docker compose, or in values.yaml if useing kubernetes.
+
+When models are changed, CHT Sync needs to be restarted for the change to take effect; when it restarts, it automatically downloads the latest version of cht pipeline and applies than changes; for models materialized as views, they are are updated instantly, for models materialized as incmrenetal tables, the table is dropped and will be unavailable until it is rebuilt from scratch.
+
+## Base Models
+
+The base models define the data which is common to all CHT applications.
+All tables contain a uuid which is the primary key for the table; it is also the `_id` from the source couchdb document.
+
+{{< figure src="cht-pipeline-er.png" link="cht-pipeline-er.png" class=" center col-16 col-lg-12" >}}
+
+### `couchdb`
+All documents are stored in the couchdb db table; downstream models move fields from the document into fields and index them.
+It is possible to get any field from a couchdb document using this table; it is recommended for performance reasons to instead use the columns from downstream models.
+|Field|Description|
+|--|--|
+|`uuid`|CouchDB's unique identifier of the record|
+|`type`|The general type of the document, see below|
+|`doc`| JSON of the source document|
+
+### `data_record`
+All form responses are stored in the `data_record` table, see more details [here]({{< ref "core/overview/db-schema#reports" >}}).
+This table contains columns for the contact who made the report, the parent of that contact, the report data, and a copy of fields in jsonb format.
+It is not recommended to use the fields column directly, but instead to add one new model for each form, moving the relevant fields for that form into columns
+
+|Field|Description|
+|--|--|
+|`uuid`|Unique identifier of the record. Note this is both the primary key for this table, and a foreign key to the couchdb table.|
+|`contact_uuid`| uuid of the `contact` who made the report|
+|`parent_contact_uuid`| uuid of the parent of `contact` who made the report (usually a place)|
```

Comment:
Having this named as `contact_parent_uuid` as we have it in the model currently makes it clearer.